tests/foreman/cli/test_contentcredentials.py

```diff
@@ -669,223 +684,38 @@ def test_positive_update_key_for_repo_from_product_with_repos(module_org):
-    # Associate gpg key with a repository
-    Repository.update(
-        {'gpg-key-id': gpg_key['id'], 'id': repo['id'], 'organization-id': module_org.id}
-    )
-    # Verify gpg key was associated with the repository but not with the
-    # product
-    product = Product.info({'id': product['id'], 'organization-id': module_org.id})
-    repo = Repository.info({'id': repo['id']})
-    assert product['gpg']['gpg-key'] != gpg_key['name']
-    assert repo['gpg-key'].get('name') == gpg_key['name']
-    # Delete the gpg key
-    ContentCredential.delete({'name': gpg_key['name'], 'organization-id': module_org.id})
-    # Verify gpg key was actually deleted
-    with pytest.raises(CLIReturnCodeError):
-        ContentCredential.info({'id': gpg_key['id'], 'organization-id': module_org.id})
-    # Verify gpg key was disassociated from the repository
-    repo = Repository.info({'id': repo['id']})
-    assert repo['gpg-key'].get('name') != gpg_key['name']
-
-
-@pytest.mark.tier2
-def test_positive_delete_key_for_repo_from_product_with_repos(module_org):
-    """Create gpg key with valid name and valid gpg key via file
-    import then associate it to repository from custom product that has
-    more than one repository then delete the key
-
-    :id: 7d6a278b-1063-4e72-bc32-ca60bd17bb84
-
-    :expectedresults: gpg key is associated with a single repository but
-        not the product during creation and removed from repository after
-        deletion
-
-    :CaseLevel: Integration
-    """
-    # Create product, repositories and gpg key
-    product = make_product({'organization-id': module_org.id})
-    repos = []
-    for _ in range(gen_integer(2, 5)):
-        repos.append(make_repository({'product-id': product['id']}))
-    gpg_key = make_content_credential({'organization-id': module_org.id})
-    # Associate gpg key with a repository
-    Repository.update(
-        {'gpg-key-id': gpg_key['id'], 'id': repos[0]['id'], 'organization-id': module_org.id}
-    )
-    # Verify gpg key was associated with the repository
-    repos[0] = Repository.info({'id': repos[0]['id']})
-    assert repos[0]['gpg-key']['name'] == gpg_key['name']
-    # Delete the gpg key
-    ContentCredential.delete({'name': gpg_key['name'], 'organization-id': module_org.id})
-    # Verify gpg key was actually deleted
-    with pytest.raises(CLIReturnCodeError):
-        ContentCredential.info({'id': gpg_key['id'], 'organization-id': module_org.id})
-    # Verify gpg key is not associated with any repository or the product
-    # itself
-    product = Product.info({'id': product['id'], 'organization-id': module_org.id})
-    assert product['gpg']['gpg-key'] != gpg_key['name']
-    for repo in repos:
-        repo = Repository.info({'id': repo['id']})
-        assert repo['gpg-key'].get('name') != gpg_key['name']
-
-
 @pytest.mark.tier1
-def test_positive_list(module_org):
+def test_positive_list(module_target_sat, default_org):
```

Comment:
why updating test to use default_org, is list doesn't work on module_org?

Reply:
This test is flaky on module_org as it's already filed with several CCs. 
My understanding is a long list is truncated, so this test doesn't always find the result from CLI when using a sat filled with many other CCs. Perhaps we could use `cli_factory.make_organization()` for it?

Reply:
@Gauravtalreja1 Or should `module_org` return a fresh empty organization for each test?

Reply:
@damoore044 `module_org` should be the same for all tests run in a given module (e.g. cli/test_contentcredential.py). I would avoid modifying the `default_org` unless there's a specific requirement to do so, though, as any changes to `default_org` will persist outside of this module. If you need a tighter scope than module-level, using `function_org` would be the way to go.

If the issue you're trying to work around here is a truncated list, though, I would recommend not changing the use of `module_org` and instead being more targeted in your list command. Hammer allows listing specific content credentials by name (although not by ID for some reason). So if you change the list command to:
```
gpg_key_list = target_sat.cli.ContentCredential.list({'organization-id': default_org.id, 'name': gpg_key['name']}) 
```
...this should return only the named content credential in the output.

Reply:
Updated this with @synkd's recommendation to the command scope, running on module_org